### PR TITLE
Added support for network security groups, updated schema in the header

### PR DESCRIPTION
**Pull request by TheAzureGuy**

### CODE REVIEW

**Comment by singhkay** [comment]
@TheAzureGuy Actually ignore the bot for now. It should be "description". We're fixing the bot to detect this

**Comment by TheAzureGuy** [comment]
It is actually painful to submit a PR containing changes in multiple templates. Is this by design?

**Comment by johngossman** [comment]
@TheAzureGuy Yeah, the bot needs some work

File: mongodb-high-availability/azuredeploy.json [hunk 1]
```diff
@@ -1,65 +1,65 @@
-{
+{ 
     "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
     "contentVersion": "1.0.0.0",
     "parameters": {
         "adminUsername": {
             "type": "string",
             "metadata": {
-                "description": "Administrator user name used when provisioning virtual machines (which also becomes a system user administrator in MongoDB)"
+                "Description": "Administrator user name used when provisioning virtual machines (which also becomes a system user administrator in MongoDB)"
```
Comment (johngossman) [inline mongodb-high-availability/azuredeploy.json]: Leave it as "description". We'll fix the bot.

File: mongodb-high-availability/azuredeploy.json [hunk 2]
```diff
@@ -1,65 +1,65 @@
...
         },
         "storageAccountName": {
             "type": "string",
             "defaultValue": "",
             "metadata": {
-                "description": "Unique namespace for the Storage Account where the Virtual Machine's disks will be placed (this name will be used as a prefix to create one or more storage accounts as per t-shirt size)"
+                "Description": "Unique namespace for the Storage Account where the Virtual Machine's disks will be placed (this name will be used as a prefix to create one or more storage accounts as per t-shirt size)"
             }
         },
         "region": {
```
Comment (johngossman) [inline mongodb-high-availability/azuredeploy.json]: Total nit, but location is better. Now that we can use these templates in Azure Pack as well as Azure, region doesn't make as much sense.

**Comment by TheAzureGuy** [comment]
Submitted a new PR with changes as per feedback

**Pull request closed without merging**